src/SpriteKit/SKUniform.cs

```diff
@@ -0,0 +1,220 @@
+//
+// SKUniform.cs: SKUniform class
+//
+// Authors:
+//   Vincent Dondain (vidondai@microsoft.com)
+//
+// Copyright 2016 Xamarin Inc.
+//
+
+using System;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+using Vector2 = global::OpenTK.Vector2;
+using Vector3 = global::OpenTK.Vector3;
+using Vector4 = global::OpenTK.Vector4;
+using Matrix2 = global::OpenTK.Matrix2;
+using Matrix3 = global::OpenTK.Matrix3;
+using Matrix4 = global::OpenTK.Matrix4;
+
+#if XAMCORE_2_0 || !MONOMAC
+namespace XamCore.SpriteKit {
+	public partial class SKUniform {
+
+		static bool? versionCheck = null;
+
+		static bool CheckSystemVersion ()
+		{
+			if (!versionCheck.HasValue) {
+#if MONOMAC
+				versionCheck = PlatformHelper.CheckSystemVersion (10, 12);
+#else
+				versionCheck = XamCore.UIKit.UIDevice.CurrentDevice.CheckSystemVersion (10, 0);
+#endif
+			}
+			return versionCheck.Value;
+		}
+
+		// Apple deprecated initWithName:floatVector2: in macOS10.12/iOS10.0
+		// and made available initWithName:vectorFloat2: so we invoke
+		// the right one at runtime depending on which OS version we are running
+		public SKUniform (string name, Vector2 value)
+		{
+			if (CheckSystemVersion ())
+				Handle = InitWithNameVectorFloat2 (name, value);
```

Comment:
@rolfbjarne fix in https://github.com/xamarin/xamarin-macios/pull/498/commits/033d3b8fe1d4ee2ef25f55f2127c4bb7c9c5b9aa